### ISSUE TITLE
您好，我现在想做一下电商评论的情感分析文本分类，我数据集是自己爬虫制作的二分类数据集，如果我需要运用您的这个GCN去做二分类（好评，差评）可以去替换成我自己制作的数据集吗？因为我看您在文中测试的MR数据集也是关于电影评论的二分类，所以我想问一下，如果我用自己的数据集改把数据集放在哪呢，谢谢啦）

### ISSUE THREAD

**Issue by wangzhedaye** [yao8839836/text_gcn#52]

**Comment by yao8839836** [comment]
@wangzhedaye 

您好，请查看data/mr.txt以及data/corpus/mr.txt的格式 ，可以直接替换这两个文件。或者放在跟他们同样的位置，并修改命令行参数。

**Comment by wangzhedaye** [comment]
我看了一下，这两个文件夹是对英文电影评论的英语评价，如果我是中文的，是不是需要修改一下，或者对中文做提前的预处理呢

**Comment by yao8839836** [comment]
@wangzhedaye 

是的。中文需要先分词，以空格分隔，还可以去停用词。用以空格分隔的中文单词序列替换data/corpus/mr.txt中的英文即可，一行是一个文档。

**Comment by wangzhedaye** [comment]
我就是可以先用结巴处理自己的中文数据，然后我替换成自己的数据集了，怎么只对自己的数据集运行所有的对比模型呢，就是用自己的数据集去复现。方便可以加个联系方式么，请教您一QQ1984982367

**Comment by yao8839836** [comment]
@wangzhedaye 
Text GCN 替换了文件就可以，baseline这里没有实现，我都是从别人论文里找的代码链接或者找Github的高赞代码。

可参考 
https://github.com/yao8839836/text_gcn/issues/38

**Comment by wangzhedaye** [comment]
就是图模型可以用我处理完的数据替换就行了，但是其他的baseline得找代码，可以给个比较靠谱的链接么？或者分享一下，谢谢啦

**Comment by yao8839836** [comment]
@wangzhedaye 

除了刚才的链接#38， TFIDF我用的是sklearn工具，CNN， RNN:  https://github.com/gaussic/text-classification-cnn-rnn

pv-dbow, PV-DM: https://github.com/yao8839836/doc2vec_java

fastText和PTE直接在github搜索即可。

**Comment by wangzhedaye** [comment]
好的，谢谢啦，我去看看，非常谢谢

**Comment by VolantBoy** [comment]
很好奇楼上的大哥实验做的咋样，使用textgcn在您的数据集上的性能

**Comment by wangzhedaye** [comment]
我没有尝试textgcn了，我现在准备尝试bert，现在可能需要做细粒度的三分类

**Comment by VolantBoy** [comment]
那您没有textgcn在自己数据上的实验结果吗？我将textgcn改在了自己的数据上（中文的句对分类实验），性能很差

**Comment by wangzhedaye** [comment]
因为我后来考虑到，中文文本分类，很多语言是有前后时序性质的，textgcn可能没考虑到这些，所以效果并不是一定很好。你可以加我联系方式QQ1984982367 微信 wz1984982367 可以一起探讨一下
